main.go

```diff
@@ -6,27 +6,113 @@ import (
 	"log"
 	"net/http"
 	"regexp"
+	"strings"
 )
 
 func main() {
-	// Generate a new word for hangman
+	willRepeat := true
+	for willRepeat {
+		var (
+			word         = generateWord()
+			obscuredWord = obscureWord(word)
+			userInput    = ""
+			lives        = 6
+		)
+
+		fmt.Println("Welcome to Hangman!")
+		fmt.Printf("You have %d lives to guess the word. \n", lives)
+
+		for lives >= 0 {
+			if lives == 0 {
+				loseGame(word, &willRepeat)
```

Comment:
Why are we using pointers for `willRepeat` here? We can set `willRepeat `to false inside the for loop and break it that way. Feel like it would be cleaner to just have these functions return a bool and just use that.

Using pointers is not memory safe and could lead to a nil pointer dereference if you aren't careful. Would not recommend using them unless necessary.